aws-api/src/androidTest/java/com/amplifyframework/api/aws/TestApiCategory.java

```diff
@@ -61,7 +61,8 @@ static ApiCategory fromConfiguration(@RawRes int resourceId) throws AmplifyExcep
                 )
         );
         CategoryConfiguration apiConfiguration =
-            AmplifyConfiguration.fromConfigFile(context, resourceId)
+            AmplifyConfiguration.builder(context, resourceId)
+                .build()
```

Comment:
For these ones, I think you can keep around the factory method. Just have it proxy to the builder.

e.g.

```java
public final class AmplfiyConfiguration {
    private AmplifyConfiguration(..., ..,) { /* make one, privately */ }

    public static AmplifyConfiguration fromConfigFile(Context context, \@ResId int resourceId) {
        return builder(conext, resourceId).build();
    }

    public static Builder builder(Context context, \@ResId int resourceId) {
        reutrn new Builder(contetx, resourceId);
    }

    ...
    public static final class Builder {
        ...
        AmplifyConifguration build() { ... }
    }
}
```

If you do this, you'll cleanup the PR diff quite a bit too. There's a number of ~unnecessary changes to use the new builder, when you could just keep the factory, for most uses.

Reply:
Will do!